src/crimson/os/seastore/journal/segment_allocator.cc

```diff
@@ -144,7 +139,7 @@ SegmentAllocator::write(ceph::bufferlist to_write)
   auto write_length = to_write.length();
   auto write_start_offset = written_to;
   auto write_start_seq = journal_seq_t{
-    get_current_segment_seq(),
+    segment_seq_allocator.get_current_segment_seq(),
```

Comment:
The get_current_segment_seq() call previously relies on the journal being the only user which increments the count.  I think during roll we need to remember the seq of the current segment rather than pulling it out of the segment_seq_allocator.  In fact, I don't think there's any valid way to use SegmentSeqAllocator::get_current_segment_seq().

Reply:
Perhaps we the journal unit test could emulate calls to get_next_segment_seq() from other users in between segment rolls?